apps/dot_ext/forms.py

```diff
@@ -86,58 +107,201 @@ def clean(self):
+            "support_phone_number",
+            "logo_image",
+            "description",
+        )
+
+    def clean_name(self):
+        print("clean name")
+
+        name = self.cleaned_data.get("name")
+        print(str(name))
+        app_model = get_application_model()
+        if (
+            app_model.objects.filter(name__iexact=name)
+            .exclude(pk=self.instance.pk)
+            .exists()
+        ):
+            raise forms.ValidationError(
+                """
+                                        It looks like this application name
+                                        is already in use with another app.
+                                        Please enter a different application
+                                        name to prevent future errors.
+                                        Note that names are case-insensitive.
+                                        """
+            )
+        return name
+
+    def clean_logo_image(self):
+        print("clean logo image")
+
+        logo_image = self.cleaned_data.get("logo_image")
+        print(str(logo_image))
+        if getattr(logo_image, "name", False):
+            validate_logo_image(logo_image)
+        return logo_image
+
+    def clean_redirect_uris(self):
+        print("clean redirect uris")
+
+        redirect_uris = self.cleaned_data.get("redirect_uris")
+        print(str(redirect_uris))
+        if getattr(settings, "BLOCK_HTTP_REDIRECT_URIS", True):
+            if redirect_uris:
+                for u in redirect_uris.split():
+                    if u.startswith("http://"):
+                        msg = _("Redirect URIs must not use http.")
+                        raise forms.ValidationError(msg)
+        return redirect_uris
+
+    def clean_require_demographic_scopes(self):
+        print("clean demo scopes")
+
+        require_demographic_scopes = self.cleaned_data.get("require_demographic_scopes")
+        print(str(require_demographic_scopes))
+        if type(require_demographic_scopes) != bool:
+            msg = _(
+                "Does your application need to collect beneficary demographic information must be (Yes/No)."
+            )
+            raise forms.ValidationError(msg)
+        return require_demographic_scopes
+
+    def save(self, *args, **kwargs):
+        app = self.instance
+        print("save")
+        print(str(self.cleaned_data))
+
+        new_user_model = User.objects.create(
+            username=self.cleaned_data.get("name") + "@example.com",
+            password=str(uuid.uuid4()),
+            is_active=True,
+        )
+        group = Group.objects.get(name="BlueButton")
+        new_user_model.groups.add(group)
+        new_user_model.save()
+
+        UserProfile.objects.create(
+            user=new_user_model,
+            organization_name=self.cleaned_data.get("organization_name"),
+        )
+
+        app = super().save(*args, **kwargs)
+        app.agree = True
```

Comment:
I think we should set `app.active = False` so that the new app doesn't show up in the applications list until finally approved.

Reply:
Hmmm when I was watching Brady go through and create the new app, he set active to true (also that is what it says in the documentation). @bfausett77 could you weigh in?

Reply:
That is correct, by default we set it to true and that is also in the documentation.  If the link to get their clientId and secret expires maybe that is when we can set it to false?  I would set to true by default at this point though.

Reply:
though the current manual process leave app.active=true, setting it to false initially seems has advantages - e.g. - give reviewers (current process does not seem to have this review step;-) sufficient time to review the new app before expose to external..., this is more error proof.

Reply:
@nbragdon @bfausett77 @JFU-GIT I take that back :-)  `True` makes more sense here. For some reason was thinking there was an extra step after creation until activating.  Maybe `False` here would be good when we add in the creds sharing link part (and activate after creds are received).